src/primitives/transaction.cpp

```diff
@@ -47,8 +48,19 @@ static uint256 ComputeCMutableTransactionHash(const CMutableTransaction &tx) {
     return SerializeHash(tx, SER_GETHASH, 0);
 }
 
+static uint256 ComputeTxId(int32_t nVersion, const std::vector<CTxIn> &vin,
```

Comment:
It seems like the TxID constructor should do this operation. Is there a reason to compose it like this?